src/middleware/router.coffee

```diff
@@ -84,6 +85,8 @@ handleServerError = (ctx, err) ->
 
 sendRequestToRoutes = (ctx, routes, next) ->
   promises = []
```

Comment:
It seems that this doesn't need to be an array anymore if it is only dealing with a single primary promise. Maybe naming it primaryPromise would be better to.